modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/resource/v1_0/test/OpenAPIResourceTest.java

```diff
@@ -66,6 +73,134 @@ public void setUp() throws Exception {
 					RandomTestUtil.randomString(), _OBJECT_FIELD_NAME, false)));
 	}
 
+	@Test
+	public void testGetNestedEntityInManyToManyObjectRelationship()
+		throws Exception {
+
+		PropsUtil.addProperties(
+			UnicodePropertiesBuilder.setProperty(
+				"feature.flag.LPS-168462", "true"
+			).build());
+
+		_objectDefinition2 = ObjectDefinitionTestUtil.publishObjectDefinition(
+			Collections.singletonList(
+				ObjectFieldUtil.createObjectField(
+					"Text", "String", true, true, null,
+					RandomTestUtil.randomString(), _OBJECT_FIELD_NAME, false)));
+
+		ObjectRelationship objectRelationship =
+			ObjectRelationshipTestUtil.addObjectRelationship(
+				_objectDefinition1, _objectDefinition2,
+				TestPropsValues.getUserId(),
+				ObjectRelationshipConstants.TYPE_MANY_TO_MANY);
+
+		JSONObject jsonObject = HTTPTestUtil.invoke(
+			null, "/openapi", Http.Method.GET);
+
+		JSONArray jsonArray = jsonObject.getJSONArray(
+			_objectDefinition1.getRESTContextPath());
+
+		Assert.assertEquals(1, jsonArray.length());
+		Assert.assertEquals(
+			"http://localhost:8080/o" +
+				_objectDefinition1.getRESTContextPath() + "/openapi.yaml",
+			jsonArray.get(0));
+
+		jsonObject = HTTPTestUtil.invoke(
+			null, _objectDefinition1.getRESTContextPath() + "/openapi.json",
+			Http.Method.GET);
+
+		Assert.assertNotNull(jsonObject.getString("openapi"));
+
+		Assert.assertEquals(
+			_getNestedEntitySchema(
+				jsonObject, objectRelationship, _objectDefinition1),
+			_objectDefinition2.getShortName());
+
+		Assert.assertNull(
+			jsonObject.getJSONArray(_objectDefinition2.getRESTContextPath()));
+
+		jsonObject = HTTPTestUtil.invoke(
+			null, _objectDefinition2.getRESTContextPath() + "/openapi.json",
+			Http.Method.GET);
+
+		Assert.assertNotNull(jsonObject.getString("openapi"));
+
+		Assert.assertEquals(
+			_getNestedEntitySchema(
+				jsonObject, objectRelationship, _objectDefinition2),
+			_objectDefinition1.getShortName());
+
+		PropsUtil.addProperties(
+			UnicodePropertiesBuilder.setProperty(
+				"feature.flag.LPS-168462", "false"
+			).build());
+	}
+
+	@Test
+	public void testGetNestedEntityInOneToManyObjectRelationship()
```

Comment:
Both tests look equals to me except for the relationship
![image](https://user-images.githubusercontent.com/17163902/216351310-e7f22c15-8afa-4b03-97b9-90c766db76ab.png)

Could we avoid duplicate code?

Also, correct me if I'm wrong but Brian tell us to unify test, right? I mean, check for one-to-many and many-to-many in the same test. is that correct? I don't remember properly right now.